data_gatherer_chauffeur.py

```diff
@@ -0,0 +1,172 @@
+        'sensor.camera.depth',
+        FLAGS.carla_camera_image_width,
+        FLAGS.carla_camera_image_height,
+        transform)
+    camera_setups.append(depth_camera_setup)
+    segmented_camera_setup = pylot.simulation.utils.CameraSetup(
+        pylot.utils.FRONT_SEGMENTED_CAMERA_NAME,
+        'sensor.camera.semantic_segmentation',
+        FLAGS.carla_camera_image_width,
+        FLAGS.carla_camera_image_height,
+        transform)
+    camera_setups.append(segmented_camera_setup)
+
+    if FLAGS.chauffeur:
+        # Height calculation relies on the fact that the camera's FOV is 90.
+        top_down_location = pylot.simulation.utils.Location(1.5, 0.0, 1.4 + FLAGS.top_down_lateral_view)
+        top_down_rotation = pylot.simulation.utils.Rotation(-90, 0, 0)
+        top_down_transform = pylot.simulation.utils.Transform(
+            top_down_location, top_down_rotation)
+        top_down_segmented_camera_setup = pylot.simulation.utils.CameraSetup(
+            pylot.utils.TOP_DOWN_SEGMENTED_CAMERA_NAME,
+            'sensor.camera.semantic_segmentation',
+            FLAGS.carla_camera_image_width,
+            FLAGS.carla_camera_image_height,
+            top_down_transform,
+            fov=90)
+        camera_setups.append(top_down_segmented_camera_setup)
+
+    return camera_setups
+
+
+def create_lidar_setups():
+    lidar_setups = []
+    if FLAGS.lidar and not FLAGS.chauffeur:
+        location = pylot.simulation.utils.Location(1.5, 0.0, 1.4)
+        rotation = pylot.simulation.utils.Rotation(0, 0, 0)
+        lidar_transform = pylot.simulation.utils.Transform(location, rotation)
+        lidar_setup = pylot.simulation.utils.LidarSetup(
+            name='front_center_lidar',
+            lidar_type='sensor.lidar.ray_cast',
+            transform=lidar_transform,
+            range=5000,  # in centimeters
+            rotation_frequency=20,
+            channels=32,
+            upper_fov=15,
+            lower_fov=-30,
+            points_per_second=500000)
+        lidar_setups.append(lidar_setup)
+    return lidar_setups
+
+
+def add_perfect_tracking_component(
+        graph, carla_op):
+    if not FLAGS.perfect_tracking:
+        return []
+    ground_tracking_stream_name = 'perfect_tracker'
+    tracking_op = [
+        pylot.operator_creator.create_perfect_tracking_op(
+            graph, ground_tracking_stream_name)]
+    graph.connect([carla_op], tracking_op)
+    return tracking_op
+
+
+def main(argv):
+    # Define graph
+    graph = erdos.graph.get_current_graph()
+
+    camera_setups = create_camera_setups()
+    lidar_setups = create_lidar_setups()
+
+    # Add operator that interacts with the Carla simulator.
+    (carla_op,
+     camera_ops,
+     _) = pylot.operator_creator.create_driver_ops(
+        graph, camera_setups, lidar_setups, auto_pilot=FLAGS.carla_auto_pilot)
+
+    # Get top-down camera setup if it exists.
+    top_down_camera_setup = None
+    for cs in camera_setups:
```

Comment:
Nit: You could plug out the chauffeur camera setup out of create_camera_setups() into create_chauffeur_camera_setup() so that you don't have to filter by camera name here.

Reply:
Done!